File: 9-palindrome-number/palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x<0) return false;

        long long digit =0;
        int n = x;
        while(n>0){
            int num = n%10;
            digit = digit*10 + num;
            n = n/10;
        }

        if(digit == x) return true;
        return false;
    }
};

File: 9-palindrome-number/palindrome-number.dart
class Solution {
  bool isPalindrome(int x) {
    int n =x;

    if(x<0) return false;
    int reverse =0;
    while(n>0){
        int digit = n%10;
        reverse = reverse*10 +digit;
        n = n ~/10 ;
    }
    if(reverse == x) return true;
    return false;
  }
}